apps/minifront/src/state/swap/helpers.ts

```diff
@@ -63,22 +64,69 @@ export const sendCandlestickDataRequest = async (
     throw new Error('Asset pair equivalent');
   }
 
-  try {
-    const { data } = await dexClient.candlestickData(
+  const suppressAbort = (err: unknown) => {
+    if (signal && !signal.aborted) {
+      throw err;
+    }
+  };
+
+  const directReq = dexClient
+    .candlestickData(
       {
         pair: { start, end },
         limit,
       },
       { signal },
-    );
-    return data;
-  } catch (err) {
-    if (err instanceof Error && err.name === 'AbortError') {
-      return;
-    } else {
-      throw err;
+    )
+    .catch(suppressAbort);
+
+  const inverseReq = dexClient
+    .candlestickData(
+      {
+        pair: { start: end, end: start },
+        limit,
+      },
+      { signal },
+    )
+    .catch(suppressAbort);
+
+  const candlestickData = (await directReq)?.data ?? [];
+
+  const inverseCandlestickData = (await inverseReq)?.data ?? [];
+
+  const collated = Map.groupBy(
```

Comment:
What's the purpose of a map if it later gets converted into an array and the keys are thrown out? If there is a specific reason, please document. Also, think it would be easier to save it as an object `{ direct: ..., inverse: ...}` than an ordered array. That way you wouldn't have to use the non-null assertion below.

Reply:
commented